src/locationsettings.cpp

```diff
@@ -814,3 +839,17 @@ void LocationSettingsPrivate::writeSettings()
         }
     }
 }
+
+bool LocationSettingsPrivate::isEmergencyCallActive()
+{
+    return m_callstate == QMceCallState::Active && m_calltype == QMceCallState::Emergency;
+}
+
+void LocationSettingsPrivate::onCallStateChanged()
+{
+    if (!m_mceCallState->valid())
+        return;
+
+    m_callstate = m_mceCallState->state();
+    m_calltype = m_mceCallState->type();
```

Comment:
Any reason to have member variables for these instead of just calling getters when evaluating the state?

Reply:
No, no any particular reason. I can drop those if polling is preferred.

Reply:
Looks like on mce-qt these are in the end plain "return iValue;" type of code so the extra copies shouldn't provide much. -> Let's just call the getters when the status is needed.

Reply:
Yeah I was just looking the same that it does not create a new request but just returns internal value. On it.

Reply:
But, if we'd need the `canToggleGPS` property I think we'd need to keep track of the state in order for that to work.

Reply:
In that case it could boil down to boolean. Not sure which would be better but such canToggle & protection could be even implemented fully in the UI side as the location setting modification is couple very specific places. 

Speaking of that, there's also already the policy support for disallowing location settings changes in the UI.

Reply:
An `enum` vs. boolean? I'd take the former any day in state tracking. It is more future proof. I think the safeguard should not be only in UI but could be also there to track what is going on beneath.

Where is that policy support implemented?

Reply:
Boolean as in canToggleGPS sounds quite much like such.

Policy is sailfish-policy, living with the mdm libraries. UI side honors the policy, e.g. systemsettings / location.qml, disables the settings if the policy prevents usage.

Reply:
I really think the `enum` serves here as we might need more granularity 517f7bd029dc2e9cc22e1b2323b63db32d0a3659 (naming improvement suggestions are welcome).